src/test/java/biz/paluch/logging/gelf/intern/sender/GelfHTTPSenderTest.java

```diff
@@ -0,0 +1,85 @@
+package biz.paluch.logging.gelf.intern.sender;
+
+import biz.paluch.logging.gelf.intern.ErrorReporter;
+import biz.paluch.logging.gelf.intern.GelfMessage;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicStatusLine;
+import org.apache.http.util.EntityUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ * (c) Alekandar - https://github.com/salex89
+ */
+
+public class GelfHTTPSenderTest {
+
+
+    @Mock
+    HttpClientBuilder builder;
+    @Mock
+    CloseableHttpClient closeableHttpClient;
+    @Mock
+    CloseableHttpResponse closeableHttpResponse;
+    @Mock
+    ErrorReporter errorReporter;
+
+    @Before
+    public void prepareMocks() throws IOException {
+        MockitoAnnotations.initMocks(this);
+//        when(closeableHttpResponse.getStatusLine()).thenReturn(new BasicStatusLine(new ProtocolVersion("HTTP", 1, 1), 202, "Accepted"));
```

Comment:
This might be a good case for optional dependencies: https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html
The `mockwebserver` dependency would only be needed in the `test` scope of this project.

Ignore my remark about OkHttp, that's just an opinion. :wink: